src/cavendish_particle_tracks/_magnification_dialog.py

```diff
@@ -30,77 +38,99 @@ def __init__(self, parent=None):
         self.b2 = Fiducial()
 
-        # drop-down lists of fiducials
-        self.cbf1 = self._setup_dropdown_fiducials_combobox()
-        self.cbf2 = self._setup_dropdown_fiducials_combobox()
-        self.cbb1 = self._setup_dropdown_fiducials_combobox(back=True)
-        self.cbb2 = self._setup_dropdown_fiducials_combobox(back=True)
+        # region UI Setup
+        self.ui_setup()
+        self.cal_layer = self.create_or_retrieve_magnification_layer()
 
-        # text boxes
-        self.txf1 = QLabel(self)
-        self.txf2 = QLabel(self)
-        self.txb1 = QLabel(self)
-        self.txb2 = QLabel(self)
+    def create_or_retrieve_magnification_layer(self) -> Points:
+        if "Magnification" in self.parent.viewer.layers:
+            return self.parent.viewer.layers["Magnification"]
+        return self.parent.viewer.add_points(name="Magnification")
 
-        self.txboxes = [self.txf1, self.txf2, self.txb1, self.txb2]
+    def ui_setup(self):
+        self.setWindowTitle("Magnification")
+        # Drop-down selection of Fiducials
+        self.cmb_front1 = self._setup_dropdown_fiducials_combobox()
+        self.cmb_front2 = self._setup_dropdown_fiducials_combobox()
+        self.cmb_back1 = self._setup_dropdown_fiducials_combobox(back=True)
+        self.cmb_back2 = self._setup_dropdown_fiducials_combobox(back=True)
+        # text boxes indicating the coordinates of the fiducials
+        self.txt_f1coord = QLabel(self)  # front fiducial 1
+        self.txt_f2coord = QLabel(self)  # front fiducial 2
+        self.txt_b1coord = QLabel(self)  # back fiducial 1
+        self.txt_b2coord = QLabel(self)  # back fiducial 2
+        self.txboxes = [
+            self.txt_f1coord,
+            self.txt_f2coord,
+            self.txt_b1coord,
+            self.txt_b2coord,
+        ]
         for txt in self.txboxes:
             txt.setMinimumWidth(200)
-
-        # add coords buttons
-        self.cof1 = QPushButton("Add")
-        self.cof2 = QPushButton("Add")
-        self.cob1 = QPushButton("Add")
-        self.cob2 = QPushButton("Add")
-        self.cof1.clicked.connect(self._on_click_add_coords_f1)
-        self.cof2.clicked.connect(self._on_click_add_coords_f2)
-        self.cob1.clicked.connect(self._on_click_add_coords_b1)
-        self.cob2.clicked.connect(self._on_click_add_coords_b2)
-
-        self.bmag = QPushButton("Calculate magnification")
-        self.bmag.clicked.connect(self._on_click_magnification)
-
+        # Add selected fiducial buttons
+        self.btn_add_f1 = QPushButton("Add")
+        self.btn_add_f2 = QPushButton("Add")
+        self.btn_add_b1 = QPushButton("Add")
+        self.btn_add_b2 = QPushButton("Add")
+        self.btn_add_f1.clicked.connect(self._on_click_add_coords_f1)
+        self.btn_add_f2.clicked.connect(self._on_click_add_coords_f2)
+        self.btn_add_b1.clicked.connect(self._on_click_add_coords_b1)
+        self.btn_add_b2.clicked.connect(self._on_click_add_coords_b2)
```

Comment:
Hmm. `add_f1_marker_button`? `add_front1_fiducial_button`? Which is clearest? @palvarezc any opinion?

Reply:
Maybe `add_front1_fiducial_button`